src/extensions/default/RelatedFiles/main.js

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2018 - present Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    'use strict';
+    
+    // Load dependent modules.
+    var AppInit             = brackets.getModule("utils/AppInit"),
+        DocumentManager     = brackets.getModule("document/DocumentManager"),
+     	Filesystem          = brackets.getModule("filesystem/FileSystem"),
+        CommandManager      = brackets.getModule("command/CommandManager"),
+        Commands            = brackets.getModule("command/Commands"),
+        StringMatch         = brackets.getModule("utils/StringMatch"),
+        RelatedFilesManager = brackets.getModule("search/RelatedFilesManager"),
+        PathUtils           = brackets.getModule("thirdparty/path-utils/path-utils");
+
+    /**
+    * @private
+    * 
+    * Used to keep track of list of Related Files for a given document.
+    *
+    **/
+    var relatedFiles;
+    
+    /**
+    * Parse the current document to find out dependencies in link and script tags.
+    * Identify whether a link is for local file or is a remote link.
+    * Store all related files link in relatedFiles variable.
+    **/
+    function parseHTML()
+    {
+        var currentDocument = DocumentManager.getCurrentDocument();
+        if (window.DOMParser && currentDocument)
+        {
+			var docText = currentDocument.getText();
+            var document = new DOMParser().parseFromString(docText, "text/html");
```

Comment:
We need not create `DOMParser` instances in every `parseHTML` call. It can be created in the `RelatedFiles` constructor. `parseFromString` is a pure function and reentrant.